src/core/hle/kernel/kernel.cpp

```diff
@@ -267,20 +267,23 @@ struct KernelCore::Impl {
         }
     }
 
-    /// Creates a new host thread ID, should only be called by GetHostThreadId
-    u32 AllocateHostThreadId(std::optional<std::size_t> core_id) {
-        if (core_id) {
+    static inline thread_local u32 host_thread_id = UINT32_MAX;
+
+    /// Gets the host thread ID for the caller, allocating a new one if this is the first time
+    u32 GetHostThreadId(std::size_t core_id) {
+        if (static_cast<s32>(host_thread_id) < 0) {
             // The first for slots are reserved for CPU core threads
```

Comment:
While we're at it, probably also fix typo

```suggestion
            // First four slots are reserved for CPU core threads
```